components/scream/src/physics/p3/p3_functions_upwind_impl.hpp

```diff
@@ -26,11 +26,12 @@ ::calc_first_order_upwind_step (
   const view_1d_ptr_array<Spack, nfield>& r)
 {
   const Int kmin_scalar = ( kdir == 1 ? k_bot : k_top);
+  const Int kmax_scalar = ( kdir == 1 ? k_top : k_bot);
   Int
     kmin = kmin_scalar / Spack::n,
     // Add 1 to make [kmin, kmax). But then the extra term (Spack::n -
     // 1) to determine pack index cancels the +1.
-    kmax = ((kdir == 1 ? k_top : k_bot) + Spack::n) / Spack::n;
+    kmax = (kmax_scalar + Spack::n) / Spack::n;
```

Comment:
Found another minor bug in calc_upwind. Things were broken when dir==1 and active k-space starts mid-pack

Reply:
That's fine. This again results from isolating this routine as a lib function. In the mini-app, zeroing of the V arrays permitted omitting downstream zeroing of this sort. But it's better to harden this function as you're doing, despite the cost of likely redundant zeroing in practice.